src/near2far.cpp

```diff
@@ -334,6 +335,81 @@ void dft_near2far::save_farfields(const char *fname, const char *prefix, const v
   delete[] EH;
 }
 
+double *dft_near2far::flux(direction df, const volume &where, double resolution) {
+    /* compute output grid size etc. */
+    int dims[4] = {1,1,1,1};
+    double dx[3] = {0,0,0};
+    direction dirs[3] = {X,Y,Z};
+    int rank = 0, N = 1;
+    LOOP_OVER_DIRECTIONS(where.dim, d) {
+      dims[rank] = int(floor(where.in_direction(d) * resolution));
+      if (dims[rank] <= 1) {
+	dims[rank] = 1;
+	dx[rank] = 0;
+      }
+      else
+	dx[rank] = where.in_direction(d) / (dims[rank] - 1);
+      N *= dims[rank];
+      dirs[rank++] = d;
+    }
+
+    /* 6 x N x Nfreq array of fields in row-major order */
+    std::complex<double> *EH = new std::complex<double>[6*N*Nfreq];
+    std::complex<double> *EH_ = new std::complex<double>[6*N*Nfreq]; // temp array for sum_to_master
+
+    /* fields for farfield_lowlevel for a single output point x */
+    std::complex<double> *EH1 = new std::complex<double>[6*Nfreq];
+
+    vec x(where.dim);
+    for (int i0 = 0; i0 < dims[0]; ++i0) {
+      x.set_direction(dirs[0], where.in_direction_min(dirs[0]) + i0*dx[0]);
+      for (int i1 = 0; i1 < dims[1]; ++i1) {
+	x.set_direction(dirs[1], where.in_direction_min(dirs[1]) + i1*dx[1]);
+	for (int i2 = 0; i2 < dims[2]; ++i2) {
+	  x.set_direction(dirs[2], where.in_direction_min(dirs[2]) + i2*dx[2]);
+	  farfield_lowlevel(EH1, x);
+	  int idx = (i0 * dims[1] + i1) * dims[2] + i2;
+	  for (int i = 0; i < Nfreq; ++i)
+	    for (int k = 0; k < 6; ++k)
+	      EH_[(k * N + idx) + (6 * N * i)] = EH1[i * 6 + k];
+	}
+      }
+    }
+
+    delete[] EH1;
+    sum_to_master(EH_, EH, 6*N*Nfreq);
+    delete[] EH_;
```

Comment:
(This is all gone once you eliminate `EH` and `EH_`, since you will accumulate `F` directly from `EH1`.)